Mathlib/Topology/Connected/RealIntervals.lean

```diff
@@ -0,0 +1,560 @@
+/-
+Copyright (c) 2024 Dennis Sweeney. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dennis Sweeney
+-/
+import Mathlib.Data.Set.Card
+import Mathlib.Topology.MetricSpace.PseudoMetric
+
+/-!
+# Connected Subsets of ℝ
+
+In this file, we classify the connected subsets of ℝ.
+
+## Main Results
+
+  - `classify_connected_bounded_reals` : A bounded connected set of reals is one of:
+    - `Ioo a b` for some `a < b`,
+    - `Ioc a b` for some `a < b`,
+    - `Ico a b` for some `a < b`,
+    - `Icc a b` for some `a < b`,
+    - A singleton `{a}`.
+
+  - `classify_connected_bounded_reals_nonempty_interior` :
+    The same as above, but omitting the singleton case.
+
+  - `classify_connected_reals` : A connected set of reals is one of:
+    - The bounded connected sets of reals above,
+    - `Iio a` for some `a : ℝ`,
+    - `Iic a` for some `a : ℝ`,
+    - `Ioi a` for some `a : ℝ`,
+    - `Ici a` for some `a : ℝ`,
+    - `Set.univ : Set ℝ` (all of ℝ).
+
+  - `encard_frontier_BoundedInterval` : Bounded intervals have exactly two endpoints.
```

Comment:
Small nit: IIRC, in English you start in lower-case after a colon. (This is different in e.g. German.)